mnemosyne/migrations/e7_311_tables.py

```
@@ -0,0 +1,125 @@
+            report["added"] += 1  # type: ignore[operator]
+
+        # Indices (best-effort; IF NOT EXISTS may not be supported
+        # in all SQLite versions, matching the upstream behavior).
+        for index_ddl in _MEMORY_EVENTS_INDEX_DDL:
+            try:
+                conn.execute(index_ddl)
+                report["indices_added"] += 1  # type: ignore[operator]
+            except sqlite3.OperationalError:
+                # IF NOT EXISTS not supported in this SQLite version.
+                # Indices are best-effort; not fatal.
+                pass
```

Ruff (0.15.20)

[warning] `try`-`except` within a loop incurs performance overhead

(PERF203)